cve_bin_tool/parsers/perl.py

```diff
@@ -1,14 +1,40 @@
 # Copyright (C) 2022 Intel Corporation
 # SPDX-License-Identifier: GPL-3.0-or-later
+"""Python script containing all functionalities related to parsing of perl's cpan files."""
+import re
 
 from cve_bin_tool.parsers import Parser
 
 
 class PerlParser(Parser):
+    """Parser for perl's cpan files"""
+
     def __init__(self, cve_db, logger):
         super().__init__(cve_db, logger)
+        self.purl_pkg_type = "cpan"
+
+    def generate_purl(self, product, version, vendor, qualifier={}, subpath=None):
+        """Generates PURL after normalizing all components."""
+        # Normalize product, version, and vendor for Perl packages
+        product = re.sub(r"[^a-zA-Z0-9._-]", "", product).lower()
+        version = re.sub(r"[^a-zA-Z0-9.+-]", "", version)
+        vendor = "UNKNOWN"  # Typically, the vendor is not explicitly defined for CPAN packages
```

Comment:
Yeah, the vendor definitely isn't common here.  Or honestly in a lot of places.  I expect we'll start seeing it crop up in more package managers eventually or we may start finding more data sources that could help us add it.  One day!

Reply:
Yes you're right, I did look around on a few ways to do this. But the most simple would be to pull package-specific data sources.
Like here, for example, we would be pulling data sources which contain more information on "cpan" based packages so that we know more about vendor.

But the exact details better be left to when we work on it!